### PR TITLE
[Core] Geometrical Projection Utilities

### PR DESCRIPTION
**Pull request by philbucher**
Hi @loumalouomega 
I would like to use the projections functions from `MortarUtilities` for mapping

I made a small test for it but unfortunately the results are not as expected, the projection gives wrong results.

The case is a simple projection onto a plane triangle

Can you please take a look?

**Outlook:** I would like to move the projection-related functions to a separate class, because `mortar_utilities.h` is a quite heavy include
Plus this makes it easier to extend the functionalities related to projections
=> I have done this locally but I would like to first have a test for it to not break stuff

### CODE REVIEW

File: kratos/tests/utilities/run_tests_projection.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from KratosMultiphysics import *
```
Comment (loumalouomega) [inline kratos/tests/utilities/run_tests_projection.py]: You don't need to create this file, the tests on KratosCoreFastSuite are run automatically

**Review by loumalouomega — CHANGES_REQUESTED** [review]
We can move the projection related utilities to a different utility file. Where is the error?

File: kratos/tests/utilities/run_tests_projection.py (hunk 1 above)
Comment (philbucher) [reply kratos/tests/utilities/run_tests_projection.py]: ofc I will remove it, this is just for your convenience

**Comment by philbucher** [comment]
The error is that the tests I created fail, even though to my understanding they should not

It is a very simple projection onto a plane triangle but it gives wrong results.
Thats why I was asking you to please have a look

**Comment by loumalouomega** [comment]
Maybe the normal is not properly oriented, let me check

**Comment by philbucher** [comment]
does the sign of the normal matter?

**Comment by loumalouomega** [comment]
Could be

File: kratos/tests/utilities/test_geometrical_projection_utilities.cpp [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+using GeometryType = typename MortarUtilities::GeometryType;
+
+GeometryType::Pointer CreateTriangle2D3NForTest()
+{
+    GeometryType::PointsArrayType points;
+    points.push_back(Kratos::make_shared<NodeType>(0.04, 0.02, 0.0));
+    points.push_back(Kratos::make_shared<NodeType>(1.1, 0.03, 0.0));
+    points.push_back(Kratos::make_shared<NodeType>(1.08, 1.0, 0.0));
+
+    return GeometryType::Pointer(new Triangle2D3<NodeType>(points));
```
Comment (loumalouomega) [inline kratos/tests/utilities/test_geometrical_projection_utilities.cpp]: This is a 2D triangle, are you trying to project in 3d space or 2D, in 2D space the geometry shoul be a line, not the whole triangle

**Review by loumalouomega — CHANGES_REQUESTED** [review]

File: kratos/tests/utilities/test_geometrical_projection_utilities.cpp (hunk 2 above)
Comment (loumalouomega) [reply kratos/tests/utilities/test_geometrical_projection_utilities.cpp]: imagen
This is  representation of the geometry

Comment (loumalouomega) [reply kratos/tests/utilities/test_geometrical_projection_utilities.cpp]: This is 2D, the normal you defined is out of the plane, I don't understand

Comment (philbucher) [reply kratos/tests/utilities/test_geometrical_projection_utilities.cpp]: copy paste error :)
Should be 3D ofc. However looking into the code I am not sure if it makes a difference, no functions from the Geometry are being used, only the first point

Comment (philbucher) [reply kratos/tests/utilities/test_geometrical_projection_utilities.cpp]: I am also trying it with 3D Triangle now

The second test also does not work, and there is no geometry being used there

Comment (loumalouomega) [reply kratos/tests/utilities/test_geometrical_projection_utilities.cpp]: OK, in any case if the Z coords is 0, the expected distance is 0

**Comment by loumalouomega** [comment]
I created an independent file and added a deprecated for the old ones, working on the test

File: kratos/tests/utilities/test_geometrical_projection_utilities.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    const double x_coord = 0.325;
+    const double y_coord = 0.147;
+
+    Point point_to_proj(x_coord, y_coord, expected_proj_dist);
+
+    array_1d<double,3> dir_vector;
+    array_1d<double,3> normal_vector;
+
+    dir_vector[0] = 0.0;
```
Comment (loumalouomega) [inline kratos/tests/utilities/test_geometrical_projection_utilities.cpp:53]:
~~~cpp
    dir_vector[0] = 0.0;
    dir_vector[0] = 0.0;
    dir_vector[0] = 1.0;

    normal_vector[0] = 0.0;
    normal_vector[0] = 0.0;
    normal_vector[0] = -1.0;
~~~

I found the error, one hint, thee is something wrong in the indexes

**Comment by loumalouomega** [comment]
Done, in fact I can auto-approve me....

File: kratos/tests/utilities/test_geometrical_projection_utilities.cpp (hunk 3 above)
Comment (philbucher) [reply kratos/tests/utilities/test_geometrical_projection_utilities.cpp:53]: Sorry for this ...

**Comment by philbucher** [comment]
Thanks a lot for debugging my sloppy stuff.

In fact as mentioned in the first post I also had those things prepared locally :)

Anyway, I did some minor improvements (naming of variables, reduce number of includes), so I think this is good to go

Thanks a lot again!

**Review by loumalouomega — APPROVED** [review]
OK

**Pull request merged**